src/Home.tsx

```diff
@@ -0,0 +1,55 @@
+import './App.css'
+import { GoogleAuthProvider, inMemoryPersistence, setPersistence, signInWithPopup } from 'firebase/auth';
+import { auth } from "../firebase"
+import { useNavigate } from 'react-router-dom';
+
+export default function Home() {
+
+  // session is saved to cookie
+
+  const navigateTo = useNavigate()
+
+  async function checkAuth() {
```

Comment:
what is the purpose of this function?

Reply:
It's used to redirect the user client-side. I haven't implemented any redirecting after login.